pkg/kubectl/stop.go

```diff
@@ -337,6 +333,8 @@ func (reaper *StatefulSetReaper) Stop(namespace, name string, timeout time.Durat
 	}
 	if timeout == 0 {
 		numReplicas := ss.Spec.Replicas
+
+		// BUG: this timeout is never used.
```

Comment:
As discussed offline, looks like a bug here to me. The timeout should be on the individual pod teardown; it might make sense to take the user provided timeout and pass it into the scaler code.